plugin/pkg/admission/limitranger/admission.go

```diff
@@ -561,48 +525,39 @@ func PodValidateLimitFunc(limitRange *corev1.LimitRange, pod *api.Pod) error {
 
 		// enforce pod limits on init containers
 		if limitType == corev1.LimitTypePod {
-			containerRequests, containerLimits := []api.ResourceList{}, []api.ResourceList{}
-			for j := range pod.Spec.Containers {
-				container := &pod.Spec.Containers[j]
-				containerRequests = append(containerRequests, container.Resources.Requests)
-				containerLimits = append(containerLimits, container.Resources.Limits)
+			v1Pod := &corev1.Pod{}
+			// these conversion functions should never error
+			if err := k8s_api_v1.Convert_core_Pod_To_v1_Pod(pod, v1Pod, nil); err != nil {
+				errs = append(errs, err)
+				return utilerrors.NewAggregate(errs)
 			}
-			podRequests := sum(containerRequests)
-			podLimits := sum(containerLimits)
-			for j := range pod.Spec.InitContainers {
-				container := &pod.Spec.InitContainers[j]
-				// take max(sum_containers, any_init_container)
-				for k, v := range container.Resources.Requests {
-					if v2, ok := podRequests[k]; ok {
-						if v.Cmp(v2) > 0 {
-							podRequests[k] = v
-						}
-					} else {
-						podRequests[k] = v
-					}
-				}
-				for k, v := range container.Resources.Limits {
-					if v2, ok := podLimits[k]; ok {
-						if v.Cmp(v2) > 0 {
-							podLimits[k] = v
-						}
-					} else {
-						podLimits[k] = v
-					}
-				}
+
+			// pod hasn't scheduled so we don't care about InPlacePodVerticalScalingEnabled
+			opts := resourcehelper.PodResourcesOptions{}
```

Comment:
It looks like this is switching to include PodOverhead by default... I'm not sure what that will do on updates of existing pods with overhead specified which pass the LimitRange without considering the overhead and fail LimitRange considering the overhead. Is that tested?

Reply:
I convinced myself it was more correct and wouldn't matter since it was for admission. I'll just add a TODO and revert changes to this file per your other comment.